crates/core_simd/examples/nbody.rs

```diff
@@ -143,14 +143,14 @@ mod nbody {
         let mut i = 0;
         for j in 0..N_BODIES {
             for k in j + 1..N_BODIES {
-                let f = r[i] * mag[i];
-                bodies[j].v -= f * bodies[k].mass;
-                bodies[k].v += f * bodies[j].mass;
+                let f = r[i] * Simd::splat(mag[i]);
+                bodies[j].v -= f * Simd::splat(bodies[k].mass);
+                bodies[k].v += f * Simd::splat(bodies[j].mass);
                 i += 1
             }
         }
         for body in bodies {
-            body.x += dt * body.v
+            body.x += Simd::splat(dt) * body.v
```

Comment:
Look how they massacred my boy.

Reply:
yeah it's not great, but needs must

Reply:
SIMD is all about performance, I know we can all get protective of our code, but you realize you're representing a larger community in both how you represent yourself and how performant your code is.  I don't think you've demonstrated either with this.  Hope Rust core fixes this.

Reply:
This is not a performance issue at all, so I'm not sure what you mean.

Reply:
@pdx-trader I am not sure where this statement of yours is actually coming from. There should be no actual change in Rust executable performance from this branch's changes, except for a significant positive change in compiler performance that was garnered when this branch was synced into std in https://github.com/rust-lang/rust/pull/91484

I made the original remark (a quote from The Godfather, issued by Vito Corleone played by Marlon Brando, after glancing at a badly mangled corpse) purely because with these changes, the library has become more unwieldy. There is obviously no question as to whether more syntactic noise has been introduced here. But while ease of use of the library might not be directly related to performance, it eventually is linked: if people can't use it proficiently and with relatively low friction then they won't write explicitly vectorized code as often. That is why I chose to express regret, in an... idiosyncratic manner. This syntax is quite natural for lots of programmers who write a lot of numerical code.

And it is not my nor Caleb's assessment that our library's performance is significantly less than that of other high-level SIMD APIs that run on the CPU (it may be less than custom-rolled assembly for a specific program, but it should be faster than e.g. kernels written as library routines because our library, unlike most C++ code, will be inlined and monomorphized for a specific program), but that is precisely why we have examples like this one to try to test that theory, and also give us some code samples to reflect on how we could improve.

I am happy to entertain further questions if you are confused on this subject, or anything I say. I know I am prone to bantersome commentary, and also that I view things in a way others do not, so I am used to having to explain things multiple times. However, my occasional burst of levity or off-beat-sounding remark does not mean I lack competence, and explaining things often requires participation from the inquirer as well. Namely, I have difficulty answering questions that aren't posed as questions, but rather as seemingly aimless critique.